### PR TITLE
ED-2098 feat:copy question set api integration

### PR DESCRIPTION
**Pull request by princegupta1131**
# SunbirdEd - Portal
---

### Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
---

#### Example
- [x] Applicable
- [ ] Not applicable

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/client/src/app/modules/core/services/copy-content/copy-content.service.ts [hunk 1]
```diff
@@ -80,12 +75,11 @@ export class CopyContentService {
           })
         );
       }));
-    }
   }
   /**
    * @since - 1.#SH-66 || 2.#SH-362
    * @param  {ContentData} contentData
-   * @description - API to copy a textbook as a course.
+   * @description - API to copy a textbook as a course & question set.
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/core/services/copy-content/copy-content.service.ts:82]: please do not change a method which says copy as course
create a new method and call that

File: src/app/routes/contentRoutes.js [hunk 2]
```diff
@@ -82,17 +82,13 @@ module.exports = (app) => {
                 }
             })
         )
-        app.all('/content/questionset/v1/copy/:id',
+    app.all('/content/questionset/v1/copy/:id',
```
Comment (rajeevsathish) [inline src/app/routes/contentRoutes.js:85]: please add logger

File: src/app/client/src/app/modules/core/services/copy-content/copy-content.service.ts [hunk 3]
```diff
@@ -53,22 +53,17 @@ export class CopyContentService {
...
-    return this.copyQuestionSet(contentData); 
-    }
-    else {
       return this.userService.userOrgDetails$.pipe(mergeMap(data => { // fetch user org details before copying content
         this.frameworkService.initialize();
         return this.formatData(contentData).pipe(
           switchMap((param: any) => {
             const option = {
-              url: this.config.urlConFig.URLS.CONTENT.COPY + '/' + contentData.identifier,
+              url: contentData?.mimeType === 'application/vnd.sunbird.questionset' ? this.config.urlConFig.URLS.QUESTIONSET.COPY + '/' + contentData.identifier : this.config.urlConFig.URLS.CONTENT.COPY + '/' + contentData.identifier,
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/core/services/copy-content/copy-content.service.ts:66]: this can be optimised

**Review by rajeevsathish — CHANGES_REQUESTED** [review]

File: src/app/routes/contentRoutes.js (hunk 2 above)
Comment (rajeshkumaravel) [reply src/app/routes/contentRoutes.js:85]: @princegupta1131 
Please avoid using `app.all`, specify what is the request type for the API
Ex:
app.post -> for post method
app.get -> for get method
CC: @rajeevsathish

**Review by rajeshkumaravel — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/core/services/copy-content/copy-content.service.ts (hunk 3 above)
Comment (princegupta1131) [reply src/app/client/src/app/modules/core/services/copy-content/copy-content.service.ts]: optimised and added the lodesh

File: src/app/routes/contentRoutes.js [hunk 4]
```diff
@@ -82,17 +82,16 @@ module.exports = (app) => {
...
         proxy(contentURL, {
             limit: reqDataLimitOfContentUpload,
             proxyReqOptDecorator: proxyUtils.decorateRequestHeaders(contentURL),
             proxyReqPathResolver: (req) => {
-                console.log('I am here and trying to fix this issue')
-                console.log(
-                    '/content/questionset/v1/copy/:id called'
-                );
+                logger.info({
+                    msg: '/content/questionset/v1/copy/:id called'
```
Comment (rajeshkumaravel) [inline src/app/routes/contentRoutes.js:93]: @princegupta1131 
please use logger with req url
ex: https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-5.2.0/src/app/routes/learnerRoutes.js#L94

CC: @rajeevsathish

**Review by rajeshkumaravel — CHANGES_REQUESTED** [review]

**Review by rajeshkumaravel — APPROVED** [review]

**Pull request merged**